testutils/storage_client.go

```diff
@@ -0,0 +1,95 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package testutils // import "go.opentelemetry.io/collector/testutils"
+
+import (
+	"context"
+	"errors"
+	"sync"
+
+	"go.opentelemetry.io/collector/extension/experimental/storage"
+)
+
+// NewMockStorageClient creates a new instance of MockStorageClient
+func NewMockStorageClient() storage.Client {
+	return &MockStorageClient{
+		st: map[string][]byte{},
+	}
+}
+
+// MockStorageClient is a mocked in memory storage client designed to be used in tests
+type MockStorageClient struct {
+	st           map[string][]byte
+	mux          sync.Mutex
```

Comment:
If the `mutex` is protecting `st`, then I would suggest reordering these so that it is:

```golang
	mu          sync.Mutex
	st           map[string][]byte
```

Also, I wouldn't us `mux` since that is a common name for http routers and could be confusing, its typical shortening is `mu`